### PR TITLE
Fix possible mod zero in fuzzConstant

### PR DESCRIPTION
**Pull request by pramodsatya**
When aggregation fuzzer is run with a `batch_size` of 0, the inner vector size calculation
in `fuzzConstant` results in mod zero:
```
auto innerVectorSize = rand<uint32_t>(rng_) % opts_.vectorSize + 1;
```

This can be reproduced by running
```
velox_aggregation_fuzzer_test --seed 1077100643 --batch_size=0 --duration_sec=60 --logtostderr=1 --minloglevel=0
```

Aggregate fuzzer is also extended to test with empty input vectors, to resolve https://github.com/facebookincubator/velox/issues/7029.

### CODE REVIEW

File: velox/vector/fuzzer/VectorFuzzer.cpp [hunk 1]
```diff
@@ -431,7 +431,7 @@ VectorPtr VectorFuzzer::fuzzConstant(const TypePtr& type, vector_size_t size) {
   // generated inner vector.
 
   // Inner vector size can't be zero.
-  auto innerVectorSize = rand<uint32_t>(rng_) % opts_.vectorSize + 1;
```
Comment (aditi-pandit) [inline velox/vector/fuzzer/VectorFuzzer.cpp:434]: @pramodsatya : It would be better to use something like (opts_.vectorSize + 1) to ensure we avoid mod 0, but still retain some randomness. But that assumes that vectorSize cannot be -ve. We should validate that wherever vectorSize is initialized.

**Review by aditi-pandit — COMMENTED** [review]
Thanks for fixing this.

File: velox/vector/fuzzer/VectorFuzzer.cpp [hunk 2]
```diff
@@ -431,7 +431,7 @@ VectorPtr VectorFuzzer::fuzzConstant(const TypePtr& type, vector_size_t size) {
...
+  auto innerVectorSize = rand<uint32_t>(rng_) % (opts_.vectorSize + 1) + 1;
```
Comment (aditi-pandit) [inline velox/vector/fuzzer/VectorFuzzer.cpp:434]: Nit : Add a comment here saying that vectorSize could be 0 so adding 1 to avoid mod 0.

**Review by aditi-pandit — APPROVED** [review]
Thanks @pramodsatya

**Comment by aditi-pandit** [comment]
@kagamiori : Please can you help review and merge.

**Comment by aditi-pandit** [comment]
@pramodsatya : Please can you rebase this code.

File: velox/exec/fuzzer/AggregationFuzzerBase.h [hunk 3]
```diff
@@ -164,6 +164,7 @@ class AggregationFuzzerBase {
   static VectorFuzzer::Options getFuzzerOptions(
       VectorFuzzer::Options::TimestampPrecision timestampPrecision) {
     VectorFuzzer::Options opts;
+    VELOX_USER_CHECK_GE(FLAGS_batch_size, 0);
```
Comment (kagamiori) [inline velox/exec/fuzzer/AggregationFuzzerBase.h:167]: Since this PR already fix the mod zero issue, do we still not allow batch_size to be 0?

There is a batch_size flag in RowNumberFuzzer too.

**Review by kagamiori — COMMENTED** [review]
Hi @pramodsatya, thank you for fixing this issue! I left a question.

File: velox/exec/fuzzer/AggregationFuzzerBase.cpp [hunk 4]
```diff
@@ -270,7 +270,11 @@ std::vector<RowVectorPtr> AggregationFuzzerBase::generateInputData(
     generator = findInputGenerator(signature.value());
   }
 
-  const auto size = vectorFuzzer_.getOptions().vectorSize;
+  auto size = vectorFuzzer_.getOptions().vectorSize;
+  // 40% of times test aggregation fuzzer on empty input vectors.
+  if (vectorFuzzer_.coinToss(0.4)) {
+    size = 0;
+  }
```
Comment (kagamiori) [inline velox/exec/fuzzer/AggregationFuzzerBase.cpp:277]: Why do we need this change? `vectorFuzzer_.getOptions().vectorSize` comes from FLAGS_batch_size, so we don't need to randomize it again.

File: velox/exec/fuzzer/AggregationFuzzerBase.h (hunk 3 above)
Comment (kagamiori) [reply velox/exec/fuzzer/AggregationFuzzerBase.h:167]: Hi @pramodsatya, thank you for taking a look. Have you tried running the aggregation fuzzer, window fuzzer, join fuzzer, and row number fuzzer with --bathc_size=0? If they all work properly, I think we can avoid `VELOX_USER_CHECK_GE(FLAGS_batch_size, 0)` for all of them. If some of these fuzzers do not work with --batch_size=0, let's add the VELOX_USER_CHECK_GE and document what is needed for them to allow it.

File: velox/exec/fuzzer/AggregationFuzzerBase.cpp (hunk 4 above)
Comment (pramodsatya) [reply velox/exec/fuzzer/AggregationFuzzerBase.cpp:277]: The batch_size flag is not modified in the Github fuzzer workflow and the default value of 100 is used for all fuzzer tests. The size is set to 0 here so the aggregate fuzzer is tested on empty input so the issue https://github.com/facebookincubator/velox/issues/7029 is resolved, since the changes from https://github.com/facebookincubator/velox/issues/7029 are included in this PR. Could you please share if this is fine, or whether it should be set randomly for fuzzer tests in `.github/workflows/experimental.yml`?

File: velox/exec/fuzzer/AggregationFuzzerBase.h (hunk 3 above)
Comment (pramodsatya) [reply velox/exec/fuzzer/AggregationFuzzerBase.h]: Thanks @kagamiori, I ran aggregate fuzzer and window fuzzer with `--batch_size=0` and don't see any errors. Join fuzzer and row number fuzzer expect the batch_size to be greater than 10, so removed the check `VELOX_USER_CHECK_GE(FLAGS_batch_size, 0)` for all of them.

File: velox/exec/fuzzer/AggregationFuzzerBase.cpp (hunk 4 above)
Comment (kagamiori) [reply velox/exec/fuzzer/AggregationFuzzerBase.cpp:277]: Hi @pramodsatya, I discussed with @kgpai. We think it may be better to make it a unit test for #7100. Please see the comment here: https://github.com/facebookincubator/velox/issues/7029#issuecomment-2111350383.
